code/modules/power/supermatter/supermatter.dm

```diff
@@ -398,7 +408,10 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		power = max( (removed.temperature * temp_factor / T0C) * gasmix_power_ratio + power, 0) //Total laser power plus an overload
 
 		if(prob(50))
-			radiation_pulse(src, power * (1 + power_transmission_bonus/10))
+			radiation_pulse(src, power * (1 + (tritiumcomp*3)*(power_transmission_bonus/(10-(bzcomp*5)))))	// Emit upto 5x the radiation at 100% BZ & 3x Rad amount at 100% Tritum
+		if(prob(10 * (bzcomp * 3)) && bzcomp > 0.4)
```

Comment:
suuuuper minor micro optimization but short-circuiting means you should probably put the bzcomp check before the prob call

Reply:
I wrote it to work not to be efficient though I'll keep it in mind

Reply:
this ain't criticism this is idle thoughts while reading code, i ain't a maintainer you can ignore me

Reply:
I know